backend/fittoring/src/main/java/fittoring/application/notification/service/MemberFcmTokenService.java

```diff
@@ -0,0 +1,46 @@
+package fittoring.application.notification.service;
+
+import fittoring.application.exception.BusinessErrorMessage;
+import fittoring.application.exception.MemberNotFoundException;
+import fittoring.application.member.repository.MemberRepository;
+import fittoring.application.notification.repository.MemberFcmTokenRepository;
+import fittoring.domain.model.MemberFcmToken;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class MemberFcmTokenService {
+
+    private final MemberFcmTokenRepository memberFcmTokenRepository;
+    private final MemberRepository memberRepository;
+
+    @Transactional
+    public void upsertFcmToken(Long memberId, String token) {
+        Optional<MemberFcmToken> tokenOptional = memberFcmTokenRepository.findByMemberId(memberId);
+        if (tokenOptional.isEmpty()) {
+            saveNewFcmToken(memberId, token);
+            return;
+        }
+        renewFcmToken(tokenOptional.get(), token);
```

Comment:
- 분기문을 람다로 깔끔하게 리팩터링 할 수 있을 것 같아요!
- `@Transactional` 걸려있으니 JPA 더티체킹으로 `save(fcmToken)`를 명시적으로 호출하지 않아도 될 것 같아요!
```suggestion
        memberFcmTokenRepository.findByMemberId(memberId).ifPresentOrElse(
                foundToken -> foundToken.updateToken(token),    // 존재하면 업데이트
                () -> saveNewFcmToken(memberId, token)  // 없으면 새로 발급
        );
```